### ISSUE TITLE
Handling degeneracies

### ISSUE THREAD

**Issue by ramcdona** [tomilov/quickhull#14]
Thanks for all your work and for making this convex hull library available.  I am evaluating it for use in my program.

My data sets will often contain a bunch of co-planar points.  In my first tests with your tool, these cause `if ( !qh.check() )` to fail.  In your example, this test is accompanied with the comment `// resulted structure is not convex (generally due to precision errors)`.

If I go ahead and try to make use of the resulting mesh, I get something messy like the following:
![ConvHullWingtip](https://github.com/tomilov/quickhull/assets/1329600/d03a08f6-ea18-40fe-b0e9-d1de4f398e97)

Is convhull expected to work in cases like this?  Does convhull use robust geometric predicates for things like the 3D orientation test?

I have attached a points file corresponding to this test case.

[test.txt](https://github.com/tomilov/quickhull/files/13781863/test.txt)

**Comment by tomilov** [comment]
No, currently it is not expected the code to work in cases like yours. Stable, robust calculations are difficult for the whole algorithm to maintain.
